### ISSUE TITLE
html code display on form after upgrade GLPI 9.0.5 to 10.0.3 

### ISSUE THREAD

**Issue by chumnor1** [pluginsGLPI/formcreator#3200]
Bug: Formcreator display error after upgrade GLPI version from 9.0.5 to 10.0.3
Noted: Error only form from glpi old version, for currently glpi ( 10.0.3 ) is fine to go.
            formcreator plugin version: 2.13.3 , tried upgrade to 2.13.4 and 2.13.5-dev version but issue still persists

![attach 1](https://user-images.githubusercontent.com/29721874/223909591-af96613a-c0e8-48ca-9228-ad024aab8c88.jpg)

**Comment by btry** [comment]
Hi

Please check this closed issue: https://github.com/pluginsGLPI/formcreator/issues/3171

**Comment by btry** [comment]
The filename and the path are wrong. You might be lost if you don't understand french.


- Remove the file you added 
-  download FixHtmlEncodingCommand.php from [this link]
(https://raw.githubusercontent.com/btry/glpi/2eaa775bb2c4184c3c685e6bcd77af1a306ef7cd/src/Console/Database/FixHtmlEncodingCommand.php)
- copy it in glpi/src/Console/Database/ folder, keep the name as is.This is the version of the file as it is after applying the PR https://github.com/glpi-project/glpi/pull/13844 for GLPI.
- run bin/console db:fix_html --dump /tmp/sql-rollback.sql

The command will identify items with bad HTML encoding, backup the current content of the fields to fix, then actually fix them. Check the altered items in GLPI. If they display properly then you're done. If not, the rollback file will help you to restore the fields altered by the command.

**Comment by chumnor1** [comment]
i also tried this and below error appear 
![error](https://user-images.githubusercontent.com/29721874/224010731-d0aad8d5-fe1f-47ad-8b96-21d18d69fbae.png)

**Comment by btry** [comment]
You must change your current directory. Go in the root folder of GLPI.

**Comment by chumnor1** [comment]
I am on GLPI directory and tried the command, then the message error appear.
![error](https://user-images.githubusercontent.com/29721874/224086958-b1f11340-ea73-4312-ba57-beba4e119c13.png)

**Comment by btry** [comment]
Still strange to see a warning about gettext module.

You're running glpi 10.0.3, and the file has been done later. There is probably a dependency with glpi 10.0.4 or later. 

I cannot check it right now. 

Could you consider upgrading to the latest glpi, as a prerequisite to use the fix ?

**Comment by chumnor1** [comment]
I am on testing upgrade to glpi 10.0.6 but seem another plugin ( fusioninventory ) work not well  on that version. and also formcreator plugin html code display issue still persist.

**Comment by btry** [comment]
It is expected that the issue with old tickets persists. You need to run the fix command.

About fusion inventory, you may consider to switch to glpi inventory.

**Comment by chumnor1** [comment]
OK, thank for your feedback, let my try it and update later.

Thank you.

**Comment by chumnor1** [comment]
hi, 

Now i updated to GLPI 10.0.6 and formcreator to version 2.13.4.

after updated, issue still persist even applied this method: https://github.com/pluginsGLPI/formcreator/issues/3200#issuecomment-1461539368

Error display:
![Screenshot 2023-03-13 174634](https://user-images.githubusercontent.com/29721874/224679947-40f55e9a-18e7-465a-8b37-5354ba6589e4.png)

**Comment by btry** [comment]
Hi

You have the plugin Generic Object. I guess that Generic Object does not declare very well the meta-data about its own tables, and the code of the fix tool  cannot handle properly this plugin.

- Take a note of all enabled plugins
- disable all plugins 
- run the tool again
- re-enable the plugins enabled in step 1

**Comment by chumnor1** [comment]
yes, after follow this instruction, it's fix for for in ticket which not yet close.
for some forms which already close ticket still error
![Screenshot 2023-03-13 174634](https://user-images.githubusercontent.com/29721874/224772279-9fdde24a-1397-43da-8cc5-48ecfa2036d6.png)

**Comment by chumnor1** [comment]
here is the form which still error
![Screenshot 2023-03-13 174634](https://user-images.githubusercontent.com/29721874/224772770-f94fefd5-7284-4ee4-8d30-8f7b99abd5d1.png)

**Comment by btry** [comment]
I think that he status if the ticket is not correlated with the presence of the issue.

I need to analyze the raw content of the column "content" in the table glpi_tickets, for the row with ID=1464. However there is may be a confidentiality issue which may prevent you to publish here a sample of your data. If you have a testing instance of GLPI, please extract the content of the ticket, create a ticket in the testing instance, then past the raw content in the newly created ticket. Try to redact any sensitive data in a way which preserves the issue. You must to it by editing the content directly in the database with your favourite MySQL tool. Once done, extract the content againt fron the testing instance, save it in a txt file then paste it here.

**Comment by chumnor1** [comment]
Hello,
Due I am on testing server and no confidential  for that data.

please check below for row data:

Ticket ID 1464  ( old ticket error display ) 

er&lt;/div&gt;&lt;div&gt;&lt;b&gt;13) Start Working Date (Join Date) : &lt;/b&gt;16-06-2021&lt;/div&gt;&lt;div&gt;&lt;b&gt;14) Last Working Date ( Case Resigning) : &lt;/b&gt;&lt;/div&gt;&lt;div&gt;&lt;b&gt;15) Account Expiry Date (For Trainee only ) : &lt;/b&gt;&lt;/div&gt;&lt;div&gt;&lt;b&gt;16) Remark : &lt;/b&gt;&lt;p&gt;Dear Bong Bunhoeung,&lt;/p&gt;&lt;p&gt;I would like to request your support on purchasing printing inks for printer device HP ColorLaserJet MFP M182-M185 at Laboratory office.&lt;/p&gt;&lt;p&gt;Regarding to request quantity, please estimated for as below qty:&lt;br /&gt;- Black : 8 pcs&lt;br /&gt;- Blue: 6 pcs&lt;br /&gt;- Yellow: 4 pcs&lt;br /&gt;- Pink: 2 pcs&lt;/p&gt;&lt;p&gt;Our department code is: DP0000123 (Laboratory/QC).&lt;/p&gt;&lt;p&gt;Shall you have any question, please let me know.&lt;/p&gt;&lt;p&gt;Thanks and Warm regards,&lt;br /&gt;Sotheany THUN&lt;br /&gt;089 227 784&lt;/p&gt;&lt;p&gt;* From Department: Supply Chain - Quality *&lt;/p&gt;&lt;/div&gt;&lt;/div&gt;

**Comment by btry** [comment]
Hi

This example seems truncated. The beginning is probably missing. Also, you should put it in a text file to avoid possible mess in the rich editor of github.

**Comment by btry** [comment]
Hi @guillermohaad 

Could you try the release 2.13.5 ?

**Comment by btry** [comment]
You have something wrong in your tables. Follow the instructions of the error message to get more the differences between the expected schema and the actual schema. You may also read release notes of the releases 2.13.x.

**Comment by btry** [comment]
Hi

Do you have feedback for this issue ?

**Comment by btry** [comment]
Hi

You very likely had an error because your tables contains errors. You just need to read the diff displayed by the command and fix the tables accordying to this diff.

**Comment by btry** [comment]
You may want to read the release notes, There are informations about that.

**Comment by btry** [comment]
I updated the instructions because you cannot enable the plugin in the current state of your setup. 

Run the following command : ```bin/console plugin:install formcreator -f```. It will ask you the login of a super admin account, then try to proceed. The database check will fail and show the differences between the expected schema and the actual schema.

**Comment by btry** [comment]
Hi

Did you fix the DB for your upgrade ?

**Comment by btry** [comment]
Hi

Please open a new ticket : context changed as you upgraded to GLPI 10.0.9 (accordying to your last comment).
